ome/ome-files-cpp
# docs: Correct and check internal links

- Adjust all doxygen links to use relative paths in the install tree
- Add a validator tool which checks the validity of all such lock links, including anchors
- Add a `sphinx-linkcheck` cmake configuration option, plus targets to validate links
- Clean up any remaining sphinx warnings

While relative links are for the locally installed version of the documentation, this will work equally well for the documentation on the download page and the main website.

--------

Testing:

- Download https://ci.openmicroscopy.org/view/Files/job/OME-FILES-CPP-DEV-merge/BUILD_TYPE=Release,node=marlin/lastSuccessfulBuild/artifact/artefacts/binaries/
- unpack it
- open share/doc/ome-files/manual/html/schema.html and check that the `model` and `metadata` links work (they fail with the previous build)
- open share/doc/ome-files/manual/html/tutorial.html and check the various links work (they previously failed)

These links should redirect to the ome-common and ome-xml and ome-files API references.  These will be `share/doc/ome-$component/api/html/$link`

The link checking will be enabled in the superbuild with an additional PR.
